### ISSUE TITLE
Optimize the payload size

### ISSUE THREAD

**Issue by jahtalab** [elastic/apm-agent-js-base#158]

**Comment by vigneshshanmugam** [comment]
One of the techniques that is used by boomerang - https://github.com/nicjansma/resourcetiming-compression.js

**Comment by vigneshshanmugam** [comment]
Yes all the measurement above are uncompressed, We are trying to optimise the Payload size that we are sending to the APM server which in turn would reduce the bandwidth consumption for the users.

**Comment by jahtalab** [comment]
Copying existing fields from the transactions to the spans on the same request. If the field exist in the span we should use it instead. 
*  `trace_id`, 
* `parent_id` 
* `transaction_id`
Having default values for fields if they are not present in the payload:
* `span.sync` default should be `false`
* `span_count` , the default should be calculated based on the number of spans
* `navigationTiming` We should add a configuration for this and also remove some of the marks that we think is not very useful.
* `start` and `duration` should use integer instead of float
* `http.url` -> we should discuss whether it would be a good solution to use the `span.name` instead (only for resource timing spans).
* Decided to add the field compression separately to see how much it will affect the rest of optimisations

**Comment by jahtalab** [comment]
We're currently block by [optimization on apm server](https://github.com/elastic/apm-server/issues/2854)

**Comment by jahtalab** [comment]
Does the `span_count` need to be the actual value for unsampled transactions? if so the agent needs to send the value for unsampled transactions.

cc @jalvz

**Comment by jalvz** [comment]
@jahtalab the original spec for `span_count` is here: https://github.com/elastic/apm-server/issues/280
